ext/curl/interface.c

```diff
@@ -2221,24 +2249,28 @@ static inline int build_mime_structure_from_hash(php_curl *ch, zval *zpostfields
 				}
 
 #if LIBCURL_VERSION_NUM >= 0x073800 /* 7.56.0 */
-				if (!(stream = php_stream_open_wrapper(ZSTR_VAL(postval), "rb", IGNORE_PATH, NULL))) {
-					zend_string_release_ex(string_key, 0);
-					return FAILURE;
-				}
+				cb_arg = emalloc(sizeof *cb_arg);
+				cb_arg->ch = ch;
+				ZVAL_COPY(&cb_arg->postfields, zpostfields);
+				zend_llist_add_element(&ch->to_free->stream, &cb_arg);
```

Comment:
This is super weird. What's going on here?

Reply:
From the commit message:

> This requires to store the zval that has been originally provided for
`CURLOPT_POSTFIELDS`.  Ideally, we would store it in an additional
member of `php_curl`, but since that would constitute a BC break, we
work around by actually storing a union in `to_free->stream`, […]

This is the code which adds the postfields to the free list. Of course, an ugly hack, but I think it's necessary for BC reasons (PHP-7.4+). For master this should be cleaned up.

Reply:
You may add the new field to the end of the struct.

Reply:
That would be great! Just pushed that.